cmd/fleet/router.go

```diff
@@ -8,34 +8,39 @@ import (
 	"fleet/internal/pkg/bulk"
 	"fleet/internal/pkg/saved"
 
+	"github.com/go-playground/validator/v10"
 	"github.com/julienschmidt/httprouter"
 )
 
 const (
 	ROUTE_ENROLL  = "/api/fleet/agents/:id"
 	ROUTE_CHECKIN = "/api/fleet/agents/:id/checkin"
 	ROUTE_ACKS    = "/api/fleet/agents/:id/acks"
+	ROUTE_RESULT  = "/api/fleet/agents/:id/result"
 )
 
 type Router struct {
-	sv     saved.CRUD
-	bulker bulk.Bulk
-	ct     *CheckinT
-	et     *EnrollerT
+	sv       saved.CRUD
+	bulker   bulk.Bulk
+	ct       *CheckinT
+	et       *EnrollerT
+	validate *validator.Validate
 }
 
 func NewRouter(sv saved.CRUD, bulker bulk.Bulk, ct *CheckinT, et *EnrollerT) *httprouter.Router {
 
 	r := Router{
-		sv:     sv,
-		bulker: bulker,
-		ct:     ct,
-		et:     et,
+		sv:       sv,
+		bulker:   bulker,
+		ct:       ct,
+		et:       et,
+		validate: validator.New(),
 	}
 
 	router := httprouter.New()
 	router.POST(ROUTE_ENROLL, r.handleEnroll)
 	router.POST(ROUTE_CHECKIN, r.handleCheckin)
 	router.POST(ROUTE_ACKS, r.handleAcks)
+	router.POST(ROUTE_RESULT, r.handleActionResult)
```

Comment:
Why the new endpoint? Why not just use the `ROUTE_ACKS`? Elastic Agent already handles actions and response with the Ack.

Reply:
It was a part of the original proposal doc and I didn't get any objections. The results endpoint is not an ack. The result document is posted upon completion of the action, with the details on of the completion an an error if error happened.
Assumed that the expiration/timeout of the action will be stored as the result document with the error code.

Reply:
So agent will both ack and providing results?

Reply:
For the agent actions (that I worked on), the agent doesn't need to ack.
The workflow for actions looks something like this:
1. The Agent calls "checkin" API
2. The Fleet Server checked the "last action token"(if provided), and sends the new actions with the "token" in response. Next time the fleet server "checks in" it sends the token to the fleet server, this updates the agent record so we track the last action received by the agent.
3. The Agent dispatches the actions to the corresponding beats/applications
4. The beats/applications execute the action and stream the data to ES directly
5. The beats/applications communicate the result/error of the action execution back to the agent upon the completion of the action
6. The Agent calls the "result" API